### PR TITLE
Update Jakarta Data TCK to 1.0.0-M1

### PR DESCRIPTION
**Pull request by KyleAure**
This PR will re-enable the Jakarta Data TCK using the 1.0.0-M1 release. 

This PR also fixes the following issues:

## TCK Runner

The TCKRunner automatically finds JARs under `wlp/dev` and adds them as properties to the Maven command so that dependencies can be resolved.

However if we have the following systemPath element: `<systemPath>${io.openliberty.jakarta.data.1.0}</systemPath>`
If a developer's environment is old and they have multiple versions of the Jakarta Data API jar like: 
```txt
wlp/dev/api/spec/
    io.openliberty.jakarta.data.1.0_1.0.80.jar
    io.openliberty.jakarta.data.1.0_1.0.81.jar
    io.openliberty.jakarta.data.1.0_1.0.82.jar
    io.openliberty.jakarta.data.1.0_1.0.83.jar
```

Then the TCK runner would set `io.openliberty.jakarta.data.1.0=.../wlp/dev/api/spec/io.openliberty.jakarta.data.1.0_1.0.80.jar` instead of the using the latest version.

## Add full tests

I added in a full tests since we now are testing CDI I wanted to make sure there was no behavior differences between the versions of CDI that included between core/web/full profile.

## Remove duplicate TCK Runner

I condensed all of the Jakarta Platform tests to use the same TCK Runner since the only difference between them was the 
protocol property for Arquillian.

### CODE REVIEW

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
#libby - copyright header check
#build - personal build

File: dev/io.openliberty.jakarta.data.1.0_fat_tck/publish/servers/io.openliberty.org.jakarta.data.1.0.full/server.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	
+    <!-- Permission needed for Oracle driver -->
+    <javaPermission className="java.lang.RuntimePermission" name="accessDeclaredMembers" />
+    
+    <!-- Permission needed for Postgres driver -->
+    <javaPermission codeBase="${server.config.dir}/apps/app.war" className="java.util.PropertyPermission" name="org.postgresql.forceBinary" actions="read"/>
+
+    <!-- Permission needed for SQLServer driver -->
+    <javaPermission className="java.util.PropertyPermission" name="java.specification.version" actions="read"/>
+    <javaPermission className="java.net.SocketPermission" name="*" actions="connect,resolve"/>
```
Comment (Azquelt) [inline dev/io.openliberty.jakarta.data.1.0_fat_tck/publish/servers/io.openliberty.org.jakarta.data.1.0.full/server.xml:59]: It looks like the first line which grants `AllPermission` makes the other entries redundant, but I guess you just  have the permissions for Oracle, Postgres and SQLServer defined in every server.xml? Either way it doesn't cause  a problem, just looks odd.

**Review by Azquelt — APPROVED** [review]
`TCKRunner` changes look good, thank you.

The rest of the changes look sensible to me, though I'm no expert on Jakarta Data.

**Comment by KyleAure** [comment]
#build - rebuild personal build forgot to upload the M1 artifact to Artifactory 😨

**Comment by KyleAure** [comment]
Personal build failed with an existing defect.

**Pull request merged**